packages/apps/plugins/plugin-status-bar/src/components/FeedbackForm.tsx

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { X, PaperPlaneTilt } from '@phosphor-icons/react';
+import React, { type FC, type PropsWithChildren, useCallback } from 'react';
+
+import { useIntentDispatcher } from '@dxos/app-framework';
+import { S } from '@dxos/echo-schema';
+import { Button, Input, Popover, useTranslation } from '@dxos/react-ui';
+import { getSize, mx } from '@dxos/react-ui-theme';
+
+import { useForm } from '../hooks';
+import { STATUS_BAR_PLUGIN } from '../meta';
+import { mkTranslation } from '../translations';
+import { validate } from '../util';
+
+const Email = S.string.pipe(
+  S.nonEmpty({ message: () => 'Email is required.' }),
+  S.pattern(/^(?!\.)(?!.*\.\.)([A-Z0-9_+-.]*)[A-Z0-9_+-]@([A-Z0-9][A-Z0-9-]*\.)+[A-Z]{2,}$/i, {
+    message: () => 'Invalid email address.',
+  }),
+);
+
+const nonEmpty = (field: string) => S.nonEmpty({ message: () => `${field} is required.` });
+const maxLength = (field: string, length: number) => S.maxLength(length, { message: () => `${field} is too long.` });
+
+// -- Types and validation.
+const FeedbackFormSchema = S.struct({
+  name: S.string.pipe(nonEmpty('Name'), maxLength('Name', 256)),
+  email: Email.pipe(maxLength('Email', 256)),
+  message: S.string.pipe(nonEmpty('Feedback'), maxLength('Feedback', 32_768)),
+});
+
+type FeedbackFormState = S.Schema.Type<typeof FeedbackFormSchema>;
+
+const initialValues: FeedbackFormState = { name: '', email: '', message: '' };
+
+const Section: FC<PropsWithChildren & { className?: string }> = ({ children, className }) => (
+  <div role='none' className={mx(className)}>
+    {children}
+  </div>
+);
+
+export const FeedbackForm = ({ onClose }: { onClose: () => void }) => {
+  const { t } = useTranslation(STATUS_BAR_PLUGIN);
+  const translation = mkTranslation(t);
+  const dispatch = useIntentDispatcher();
+
+  const onSubmit = useCallback(
+    (values: FeedbackFormState) => {
+      void dispatch({ action: 'dxos.org/plugin/observability/capture-feedback', data: values });
+      onClose();
+    },
+    [dispatch],
+  );
+
+  const { errors, handleSubmit, canSubmit, touched, getInputProps } = useForm<FeedbackFormState>({
+    initialValues,
+    validate: (values) => validate(FeedbackFormSchema, values),
+    onSubmit: (values) => onSubmit(values),
+  });
+
+  return (
+    <div role='form' className='p-3 flex flex-col gap-2'>
+      <Section className='space-b-1'>
+        <Input.Root>
```

Comment:
The `Input` set of components handles associating accessibility id matching… I think these overrides will still work, but you can just give `id` to `Input.Root` and it will generally handle everything within its subtree.

Reply:
'everthing' meaning it will set the appropriate `id`, `htmlFor`, `aria-invalid`, `aria-errormessage`, `aria-labelledby`, and `aria-describedby` attributes on all the `Input` subcomponents within its subtree.